sw/device/tests/sim_dv/spi_tx_rx_test.c

```diff
@@ -0,0 +1,391 @@
+static volatile bool exp_spi_device_irq_tx_below_level;
+static volatile bool spi_device_irq_tx_below_level_fired;
+static volatile bool exp_spi_device_irq_rx_overflow;
+static volatile bool spi_device_irq_rx_overflow_fired;
+static volatile bool exp_spi_device_irq_tx_underflow;
+static volatile bool spi_device_irq_tx_underflow_fired;
+
+/**
+ * Provides external irq handling for this test.
+ *
+ * This function overrides the default external irq handler in
+ * `sw/device/lib/handler.h`.
+ */
+void handler_irq_external(void) {
+  // Find which interrupt fired at PLIC by claiming it.
+  dif_plic_irq_id_t plic_irq_id;
+  CHECK(dif_plic_irq_claim(&plic, kTopEarlgreyPlicTargetIbex0, &plic_irq_id) ==
+            kDifPlicOk,
+        "dif_plic_irq_claim failed");
+
+  // Check if it is the right peripheral.
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralSpiDevice,
+        "Interurpt from unexpected peripheral: %d", peripheral);
+
+  // Correlate the interrupt fired at PLIC with SPI DEVICE.
+  dif_spi_device_irq_type_t spi_device_irq;
+  switch (plic_irq_id) {
+    case kTopEarlgreyPlicIrqIdSpiDeviceRxf:
+      CHECK(exp_spi_device_irq_rx_full, "Unexpected RX full interrupt");
+      spi_device_irq_rx_full_fired = true;
+      spi_device_irq = kDifSpiDeviceIrqTypeRxFull;
+      break;
+    case kTopEarlgreyPlicIrqIdSpiDeviceRxlvl:
+      CHECK(exp_spi_device_irq_rx_above_level,
+            "Unexpected RX above level interrupt");
+      spi_device_irq_rx_above_level_fired = true;
+      spi_device_irq = kDifSpiDeviceIrqTypeRxAboveLevel;
+      break;
+    case kTopEarlgreyPlicIrqIdSpiDeviceTxlvl:
+      CHECK(exp_spi_device_irq_tx_below_level,
+            "Unexpected TX below level interrupt");
+      spi_device_irq_tx_below_level_fired = true;
+      spi_device_irq = kDifSpiDeviceIrqTypeTxBelowLevel;
+      break;
+    case kTopEarlgreyPlicIrqIdSpiDeviceRxoverflow:
+      CHECK(exp_spi_device_irq_rx_overflow, "Unexpected RX overflow interrupt");
+      spi_device_irq_rx_overflow_fired = true;
+      spi_device_irq = kDifSpiDeviceIrqTypeRxOverflow;
+      LOG_INFO("rx overflow fired");
+      break;
+    case kTopEarlgreyPlicIrqIdSpiDeviceTxunderflow:
+      CHECK(exp_spi_device_irq_tx_underflow,
+            "Unexpected TX underflow interrupt");
+      spi_device_irq_tx_underflow_fired = true;
+      spi_device_irq = kDifSpiDeviceIrqTypeTxUnderflow;
+      LOG_INFO("tx underflow fired");
+      break;
+    default:
+      LOG_ERROR("Unexpected interrupt (at PLIC): %d", plic_irq_id);
+      test_status_set(kTestStatusFailed);
+      // The `abort()` call below is redundant. It is added to prevent the
+      // compilation error due to not initializing the `spi_device_irq` enum
+      // variable above. See issue #2157 for moe details.
+      abort();
```

Comment:
Instead of doing this, you can initialize spi_device_irq above to zero and avoid this problem completely.